Modules/Core/Common/include/itkNeighborhood.h

```diff
@@ -136,6 +136,11 @@ class ITK_TEMPLATE_EXPORT Neighborhood
   SizeValueType
   GetRadius(DimensionValueType n) const
   {
+    if (n >= VDimension)
+    {
+      itkExceptionMacro(<< " Can not get radius for dimension " << n << " greater than dimensionality of neighborhood "
+                        << VDimension);
+    }
     return m_Radius[n];
   }
```

Comment:
Compiler warnings like:

[Modules/Core/Common/include/itkSize.h:356](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkSize.h#L356):85: warning: array subscript 2 is above array bounds of ‘const itk::Size<2>::SizeValueType [2]’ {aka ‘const long unsigned int [2]’} [-Warray-bounds=]